### PR TITLE
Merge main into v2

### PR DESCRIPTION
**Pull request by asaj**
### Description

Merged main into v2, which mainly required integration of the latest agent changes. Please focus reviews on agent changes, specifically the scraper.

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.63"
+channel = "1.65"
```
Comment (asaj) [inline rust-toolchain:2]: Do not resolve:
- need to squash non-main commits so as not to pollute version history

**Comment by mattiecnvr** [comment]
One thing I am noticing very quickly about this diff to review is that it shows what was changed in `v2` but it does not show what was not included from `main` so while I will review this it will be very hard for me to notice if something that was in a commit from `main` was excluded when it should not have been in this merge.

File: rust/agents/scraper/src/agent.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    where
+        Self: Sized,
+    {
+        let db = ScraperDb::connect(&settings.app.db).await?;
+
+        let contract_sync_metrics = ContractSyncMetrics::new(metrics.clone());
+        let mut scrapers: HashMap<u32, SqlChainScraper> = HashMap::new();
+
+        for (chain_name, chain_setup) in settings.chain.chains.iter() {
+            // let ctx = || format!("Loading chain {}", chain_name);
```
Comment (tkporter) [inline rust/agents/scraper/src/agent.rs:48]: I'm guessing you intended to come back to this & line 50?

**Review by tkporter — APPROVED** [review]
The diff is so big that I can't say I did a very deep review- nothing drastic stands out to me though

**Review by mattiecnvr — CHANGES_REQUESTED** [review]
See review PR as well

File: rust/helm/scraper/templates/configmap.yaml [hunk 3]
```diff
@@ -6,7 +6,13 @@ metadata:
     {{- include "agent-common.labels" . | nindent 4 }}
 data:
```
[the remark this answers is not in the document]
Comment (asaj) [reply rust/helm/scraper/templates/configmap.yaml:7]: Yeah I'm planning on addressing this in a future PR

**Pull request merged**